realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -2976,4 +2977,142 @@ public void clear_all() {
         assertEquals(0, realm.where(Cat.class).count());
         assertTrue(realm.isEmpty());
     }
+
+    @Test
+    public void waitForChange_emptyDataChange() throws InterruptedException {
+        final CountDownLatch bgRealmOpenedLatch = new CountDownLatch(1);
+        final CountDownLatch bgRealmClosedLatch = new CountDownLatch(1);
+        final AtomicBoolean waitForChangeResult = new AtomicBoolean(false);
+        final AtomicLong emptyChangeCount = new AtomicLong(0);
+
+        // wait in background
+        final CountDownLatch signalTestFinished = new CountDownLatch(1);
+        Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Realm realm = Realm.getInstance(realmConfig);
+                bgRealmOpenedLatch.countDown();
+                waitForChangeResult.set(realm.waitForChange());
+                emptyChangeCount.set(realm.where(AllTypes.class).count());
+                realm.close();
+                bgRealmClosedLatch.countDown();
+            }
+        });
+        thread.start();
+
+        TestHelper.awaitOrFail(bgRealmOpenedLatch);
+        Thread.sleep(200);
```

Comment:
Please add a comment about why this is needed, just to make it easier when reading the code in the future